src/stats.rs

```diff
@@ -0,0 +1,138 @@
+use crate::config::{version, Config};
+
+use alpm::PackageReason;
+use alpm_utils::DbListExt;
+use raur::Raur;
+
+use std::cmp::Reverse;
+use std::collections::BinaryHeap;
+
+use anyhow::Result;
+use indicatif::HumanBytes;
+
+struct Info<'a> {
+    installed_packages: usize,
+    foreign_packages: usize,
+    explicit_packages: usize,
+    total_size: i64,
+    max_packages: Vec<(i64, &'a str)>,
+    orphaned: Vec<String>,
+    outdated: Vec<String>,
+}
+
+async fn collect_info<'a>(config: &'a Config, max_n: usize) -> Result<Info<'a>> {
+    let db = config.alpm.localdb();
+    let sync_db = config.alpm.syncdbs();
+
+    let installed_packages = db.pkgs().len();
+    let mut foreign_packages = Vec::new();
+    let mut explicit_packages = 0;
+    let mut total_size = 0;
+    let mut orphaned = Vec::new();
+    let mut outdated = Vec::new();
+
+    let mut max_packages = BinaryHeap::with_capacity(max_n + 1);
+
+    for pkg in db.pkgs() {
+        max_packages.push(Reverse((pkg.isize(), pkg.name())));
+        if max_packages.len() > 10 {
+            max_packages.pop();
+        }
+        if let Err(alpm::Error::PkgNotFound) = sync_db.pkg(pkg.name()) {
+            foreign_packages.push(pkg.name());
+        }
+        if pkg.reason() == PackageReason::Explicit {
+            explicit_packages += 1;
+        }
+        total_size += pkg.isize();
+    }
+
+    let aur_info = config.raur.info(&foreign_packages).await?;
```

Comment:
You can't just assume foreign package == aur package. They may be in a repo.

Reply:
I am probably not understanding you correctly. I assume that aur packages are a subset of foreign packages, and this line (50), should get me a list of installed aur packages, is this the wrong assumption?

Reply:
Yes packages may belong to a local repo. There's the two functions I mentioned in #357 but thinking about it neither of them fit perfectly. There's code in install.rs line 1684-1691 that does what you want. You should move that to a function in utils and use it. Following convention the name should be `repo_aur_pkgs` returning a list of repo and a list or aur packages.

Reply:
Thank you, it should now work as expected.

Reply:
I still see the code using a variable called foreign packages. I don't think the concept of foreign packages matters.

I think it should be:

```
Total packages:
Repo packages:
Aur packages:
```

(make sure to check the aur packages are actually in the aur)

Also with the odd and orphaned packages you should also add packages missing from the aur.

You can use `cache_info` to query all the packages into a hashmap then just iterate over the aur packages and see which ones are not in the hashmap.

There's also a Warnings struct in download.rs that you can use to do the warning printing for you.

Reply:
Also with `repo_aur_pkgs` my intention was for it to return repo and aur pkgs (hence the name). Seems as you wrote it to only return aur pkgs and the repo ones aint needed name it `aur_pkgs`.

Reply:
Yes, I misunderstood you, but It looks like I need both repo and aur packages, so I reworked the function as you intended it previously. To be completely honest, I am not really sure about this part of the code around repo packages, but I did my best to try to implement it correctly.